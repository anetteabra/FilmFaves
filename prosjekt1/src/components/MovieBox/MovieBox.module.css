.movieBox {
  width: 250px;
  margin: 10px auto;
  padding: 15px;
  border-radius: 8px; /* Gjør hjørnene avrundet */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2); /* Legg til en skygge */
  text-align: center; /* Sentrer teksten */
}

.moviePoster {
  width: 100%; /* Gjør bildet responsivt */
  border-radius: 4px;
}

.movieTitle {
  font-size: 1.5em;
  margin-top: 10px;
}

.heart {
  cursor: pointer;
  font-size: 24px; /* Juster størrelsen her */
}

.heart.fave {
  color: red;
}

.heart.notFave {
  color: grey;
}

.divdetails {
  margin-bottom: 200px;
}

/* Responsive adjustments for tablets */
@media (min-width: 600px) {
  .movieBox {
    width: 400px;
  }

  .movieTitle {
    font-size: 1.8em;
  }
}

/* Responsive adjustments for larger screens */
@media (min-width: 1024px) {
  .movieBox {
    width: 350px;
  }

  .movieTitle {
    font-size: 2em;
  }
}

/* Responsive adjustments for smaller screens (phones) */
@media (max-width: 599px) {
  .movieBox {
    width: 100%; /* Take full width on small screens */
  }

  .movieTitle {
    font-size: 1.2em;
  }
} 
