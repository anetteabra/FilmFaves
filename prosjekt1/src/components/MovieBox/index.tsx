import styles from './MovieBox.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart as solidHeart } from '@fortawesome/free-solid-svg-icons';
import { faHeart as regularHeart } from '@fortawesome/free-regular-svg-icons';

interface MovieBoxProps {
  posterPath: string;
  title: string;
  id: number;
  isFave: boolean;
  faves: number[];
  setFaves: React.Dispatch<React.SetStateAction<number[]>>;
  voteAverage: number;
  onClick?: () => void;
}

const MovieBox: React.FC<MovieBoxProps> = ({
  posterPath,
  title,
  id,
  onClick,
  isFave,
  faves,
  setFaves,
  voteAverage,
}) => {
  const toggleFave = (id: number) => {
    const newFaves = faves.includes(id)
    ? faves.filter(faveId => faveId !== id)
    : [...faves, id];
    setFaves(newFaves);
    localStorage.setItem('faves', JSON.stringify(newFaves));
  };

  return (
    <div
      onClick={onClick}
      style={{ cursor: 'pointer' }}
      className={styles.movieBox}>
      <img
        src={`https://image.tmdb.org/t/p/w500${posterPath}`}
        alt={title}
        className={styles.moviePoster}
      />
      {/* <h2 className={styles.movieTitle}>{title}</h2> */}
      <h2 className={styles.movieTitle}>Rating: {voteAverage.toFixed(1)}/10</h2>
      <span onClick={(e) => { 
        e.stopPropagation(); 
        toggleFave(id); 
      }}>
        <FontAwesomeIcon
          icon={isFave ? solidHeart : regularHeart}
          className={`${styles.heart} ${isFave ? styles.fave : styles.notFave}`}
        />
      </span>
    </div>
  );
};

export default MovieBox;
