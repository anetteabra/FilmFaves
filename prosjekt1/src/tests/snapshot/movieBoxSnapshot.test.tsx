import { render } from '@testing-library/react';
import MovieBox from '../../components/MovieBox';
import { vi } from 'vitest';

test('renders MovieBox component correctly', () => {
  const mockProps = {
    posterPath: '/path/to/poster.jpg',
    title: 'Mock Movie',
    id: 1,
    isFave: false,
    toggleFave: vi.fn(), // Mocking toggleFave
    voteAverage: 8.5,
    onClick: vi.fn(), // Mocking onClick
  };

  const movieBox = render(<MovieBox {...mockProps} />);
  expect(movieBox).toMatchSnapshot();
});
