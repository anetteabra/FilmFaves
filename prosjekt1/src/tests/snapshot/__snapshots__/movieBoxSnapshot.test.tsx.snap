// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders MovieBox component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_movieBox_aa8b66"
        style="cursor: pointer;"
      >
        <img
          alt="Mock Movie"
          class="_moviePoster_aa8b66"
          src="https://image.tmdb.org/t/p/w500/path/to/poster.jpg"
        />
        <h2
          class="_movieTitle_aa8b66"
        >
          Rating: 
          8.5
          /10
        </h2>
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart _heart_aa8b66 _notFave_aa8b66"
            data-icon="heart"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M225.8 468.2l-2.5-2.3L48.1 303.2C17.4 274.7 0 234.7 0 192.8l0-3.3c0-70.4 50-130.8 119.2-144C158.6 37.9 198.9 47 231 69.6c9 6.4 17.4 13.8 25 22.3c4.2-4.8 8.7-9.2 13.5-13.3c3.7-3.2 7.5-6.2 11.5-9c0 0 0 0 0 0C313.1 47 353.4 37.9 392.8 45.4C462 58.6 512 119.1 512 189.5l0 3.3c0 41.9-17.4 81.9-48.1 110.4L288.7 465.9l-2.5 2.3c-8.2 7.6-19 11.9-30.2 11.9s-22-4.2-30.2-11.9zM239.1 145c-.4-.3-.7-.7-1-1.1l-17.8-20-.1-.1s0 0 0 0c-23.1-25.9-58-37.7-92-31.2C81.6 101.5 48 142.1 48 189.5l0 3.3c0 28.5 11.9 55.8 32.8 75.2L256 430.7 431.2 268c20.9-19.4 32.8-46.7 32.8-75.2l0-3.3c0-47.3-33.6-88-80.1-96.9c-34-6.5-69 5.4-92 31.2c0 0 0 0-.1 .1s0 0-.1 .1l-17.8 20c-.3 .4-.7 .7-1 1.1c-4.5 4.5-10.6 7-16.9 7s-12.4-2.5-16.9-7z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_movieBox_aa8b66"
      style="cursor: pointer;"
    >
      <img
        alt="Mock Movie"
        class="_moviePoster_aa8b66"
        src="https://image.tmdb.org/t/p/w500/path/to/poster.jpg"
      />
      <h2
        class="_movieTitle_aa8b66"
      >
        Rating: 
        8.5
        /10
      </h2>
      <span>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart _heart_aa8b66 _notFave_aa8b66"
          data-icon="heart"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M225.8 468.2l-2.5-2.3L48.1 303.2C17.4 274.7 0 234.7 0 192.8l0-3.3c0-70.4 50-130.8 119.2-144C158.6 37.9 198.9 47 231 69.6c9 6.4 17.4 13.8 25 22.3c4.2-4.8 8.7-9.2 13.5-13.3c3.7-3.2 7.5-6.2 11.5-9c0 0 0 0 0 0C313.1 47 353.4 37.9 392.8 45.4C462 58.6 512 119.1 512 189.5l0 3.3c0 41.9-17.4 81.9-48.1 110.4L288.7 465.9l-2.5 2.3c-8.2 7.6-19 11.9-30.2 11.9s-22-4.2-30.2-11.9zM239.1 145c-.4-.3-.7-.7-1-1.1l-17.8-20-.1-.1s0 0 0 0c-23.1-25.9-58-37.7-92-31.2C81.6 101.5 48 142.1 48 189.5l0 3.3c0 28.5 11.9 55.8 32.8 75.2L256 430.7 431.2 268c20.9-19.4 32.8-46.7 32.8-75.2l0-3.3c0-47.3-33.6-88-80.1-96.9c-34-6.5-69 5.4-92 31.2c0 0 0 0-.1 .1s0 0-.1 .1l-17.8 20c-.3 .4-.7 .7-1 1.1c-4.5 4.5-10.6 7-16.9 7s-12.4-2.5-16.9-7z"
            fill="currentColor"
          />
        </svg>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
